File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/objects/InputObjectTest.java
package com.github.dedis.popstellar.model.objects;

import static com.github.dedis.popstellar.testutils.Base64DataUtils.generateKeyPair;
import static org.junit.Assert.assertEquals;

import com.github.dedis.popstellar.model.objects.digitalcash.ScriptInputObject;
import com.github.dedis.popstellar.model.objects.security.*;
import java.security.GeneralSecurityException;
import org.junit.Before;
import org.junit.Test;

public class InputObjectTest {
  private static final int TX_OUT_INDEX = 0;
  private static final String Tx_OUT_HASH = "47DEQpj8HBSa--TImW-5JCeuQeRkm5NMpJWZG3hSuFU=";

  private static final KeyPair SENDER_KEY = generateKeyPair();
  private static final PublicKey SENDER = SENDER_KEY.publicKey;

  private String type;
  private String pubKey;
  private String sig;
  private ScriptInputObject scriptTxIn;
  private InputObject input;

  @Before
  public void setup() throws GeneralSecurityException {
    type = "P2PKH";
    pubKey = SENDER.getEncoded();
    sig = SENDER_KEY.sign(SENDER).getEncoded();
    scriptTxIn = new ScriptInputObject(type, new PublicKey(pubKey), new Signature(sig));
    input = new InputObject(Tx_OUT_HASH, TX_OUT_INDEX, scriptTxIn);
  }

  @Test
  public void getTxOutIndexTest() {
    assertEquals(TX_OUT_INDEX, input.txOutIndex);
  }

  @Test
  public void getTxOutHashTest() {
    assertEquals(Tx_OUT_HASH, input.txOutHash);
  }

  @Test
  public void getScriptTest() {
    assertEquals(input.script.getPubKey().getEncoded(), pubKey);
    assertEquals(input.script.sig.getEncoded(), sig);
    assertEquals(input.script.type, type);
  }
}

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/model/objects/InputObjectTest.kt
package com.github.dedis.popstellar.model.objects

import com.github.dedis.popstellar.model.objects.digitalcash.ScriptInputObject
import com.github.dedis.popstellar.model.objects.security.PublicKey
import com.github.dedis.popstellar.model.objects.security.Signature
import com.github.dedis.popstellar.testutils.Base64DataUtils
import org.junit.Assert
import org.junit.Test

class InputObjectTest {
  private var type: String = "P2PKH"
  private var pubKey: String = SENDER.encoded
  private var sig: String = SENDER_KEY.sign(SENDER).encoded
  private var scriptTxIn: ScriptInputObject =
    ScriptInputObject(type, PublicKey(pubKey), Signature(sig))
  private var input: InputObject = InputObject(Tx_OUT_HASH, TX_OUT_INDEX, scriptTxIn)

  @Test
  fun txOutIndexTest() {
    Assert.assertEquals(TX_OUT_INDEX, input.txOutIndex)
  }

  @Test
  fun txOutHashTest() {
    Assert.assertEquals(Tx_OUT_HASH, input.txOutHash)
  }

  @Test
  fun scriptTest() {
    Assert.assertEquals(input.script.pubKey.encoded, pubKey)
    Assert.assertEquals(input.script.sig.encoded, sig)
    Assert.assertEquals(input.script.type, type)
  }

  companion object {
    private const val TX_OUT_INDEX = 0
    private const val Tx_OUT_HASH = "47DEQpj8HBSa--TImW-5JCeuQeRkm5NMpJWZG3hSuFU="
    private val SENDER_KEY = Base64DataUtils.generateKeyPair()
    private val SENDER = SENDER_KEY.publicKey
  }
}
